### PR TITLE
Update completion behavior

### PR DESCRIPTION
**Pull request by srusskih**
* used sync call to Jedi for completion to rid of blinking
* sync call to jedi has a timeout to keep editor from restart in case completion are sloooow
* replaced `sublime.EventListerner` -> `sublime.ViewEventListener`
* jedi requestor (Daemon) moved from `utils.py` to `daemon.py`
* jedi api interface (JediFacade) moved from `daemon.py` to `facade.py`
* improved argument completion behavior

### CODE REVIEW

**Comment by edelvalle** [comment]
I will try this, and give you some feedback

**Comment by srusskih** [comment]
@edelvalle due to I did lots of changes in `utils.py` and `daemon.py` i had to re-apply your's changes from #246. could you please review if I did it right?

File: sublime_jedi/daemon.py [hunk 1]
```diff
@@ -1,250 +1,176 @@
...
+    """
+    def _decorator(func):
+        @wraps(func)
+        def _wrapper(*args, **kwargs):
+            for window in sublime.windows():
+                if window.id() == window_id:
+                    return func(window.active_view(), *args, **kwargs)
 
-        parameters = get_function_parameters(in_call)
+            logger.info(
```
[the remark this answers is not in the document]
Comment (edelvalle) [reply sublime_jedi/daemon.py]: well it would happen in weird occasions and does not break stuff... so a warning should do.

**Comment by edelvalle** [comment]
Using goto definition fails...

```
Traceback (most recent call last):
  File "/home/xigurat/.config/sublime-text-3/Packages/Jedi - Python autocompletion/sublime_jedi/daemon.py", line 93, in _async_summon
    run_in_active_view(window_id)(callback)(answer)
  File "/home/xigurat/.config/sublime-text-3/Packages/Jedi - Python autocompletion/sublime_jedi/daemon.py", line 109, in _wrapper
    return func(window.active_view(), *args, **kwargs)
  File "/home/xigurat/.config/sublime-text-3/Packages/Jedi - Python autocompletion/sublime_jedi/go_to.py", line 143, in handle_definitions
    self._jump_to_in_window(*defn)
  File "/home/xigurat/.config/sublime-text-3/Packages/Jedi - Python autocompletion/sublime_jedi/go_to.py", line 42, in _jump_to_in_window
    flags = self.prepare_layout(active_window, transient, filename)
  File "/home/xigurat/.config/sublime-text-3/Packages/Jedi - Python autocompletion/sublime_jedi/go_to.py", line 58, in prepare_layout
    goto_layout = get_settings_param(self.view, 'sublime_goto_layout')
NameError: global name 'get_settings_param' is not defined
```

**Comment by edelvalle** [comment]
#254 Fixes this, but don't merge yet. I still evaluating...

**Comment by edelvalle** [comment]
This is ready, I had no problems with it for the past weeks!
Thanks!

**Comment by srusskih** [comment]
@edelvalle thanks for testing session! ⭐️

**Pull request merged**